cpp/net/connection_pool.cc

```diff
@@ -16,45 +17,98 @@ DEFINE_int32(url_fetcher_max_conn_per_host_port, 4,
 namespace cert_trans {
 namespace internal {
 
+ConnectionPool::Connection::Connection(evhtp_connection_t* conn,
+                                       HostPortPair&& other_end)
+    : conn_(CHECK_NOTNULL(conn)), other_end_(other_end) {
```

Comment:
Don't you need it to be `other_end_(move(other_end))` if you want to avoid the copy?